handlers/association/association_handler.go

```
@@ -0,0 +1,973 @@
+	var roleID int
+	err = tx.QueryRow("SELECT id_roles FROM roles WHERE name = $1", roleName).Scan(&roleID)
+	if err != nil {
+		utils.LogFooter()
+		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"error": "Failed to get role information",
+		})
+	}
+
+	// Safety check to avoid duplicate assignments
+	var alreadyRespo bool
+	err = tx.QueryRow("SELECT EXISTS(SELECT 1 FROM newf_roles WHERE email = $1 AND id_roles = $2)", req.Email, roleID).Scan(&alreadyRespo)
```

golangci-lint (2.12.2)

[error] ineffectual assignment to err

(ineffassign)